dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -1016,26 +1019,57 @@ private String getEmail(GHMyself myself) throws IOException {
      * Updates a user object with metadata from GitHub
      * @param user the user to be updated
      */
-    public void syncUserMetadataFromGitHub(User user) {
+    public void syncUserMetadataFromGitHub(User user, TokenDAO tokenDAO, boolean updateToken) {
```

Comment:
You only use `tokenDAO` if `updateToken` is true. And I think you have the token already whenever you invoke this method with `updateToken` set to true. So I suggest that you replace these two parameters with an Optional<Token>, and the line 1030 you test if it's an empty Optional or not.